stolostron/backlog#20720 add subctl (#3276)
open-cluster-management/rhacm-docs: services/submariner/submariner.adoc

See the https://submariner.io/getting-started/#prerequisites[Submariner upstream prerequisites documentation] for more detailed information about the prerequisites.

[#submariner-subctl]
== subctl command utility

> Submariner contains the `subctl` utility that provides additional commands that simplify running tasks on your Submariner environment. After adding the utility to your path, the following table briefly describes the available commands:

Reviewer: Do we need to explain how to install it? (We won’t be able to provide “nice” ways for OpenShift users to install `subctl` for 2.5, but perhaps we can discuss providing an RPM again later, after Submariner 0.13.)

Author: @skitt Yes, that would be good. I was hoping it was installed with ACM, and will be available after ACM is installed.

Reviewer: We make it available in a container image, I’ll check whether than can be used straightforwardly.

Author: Sounds great. Thanks. Fingers crossed.

Reviewer 2: Something like this (from Steve) is what I've seen to extract `subctl` from the container it's packaged in downstream.

```
# Download and extract subctl
oc image extract registry-proxy.engineering.redhat.com/rh-osbs/rhacm2-tech-preview-subctl-rhel8:${VERSION} --path=/dist/subctl-${VERSION}-linux-amd64.tar.xz:/tmp/ --confirm

tar -C /tmp/ -xf /tmp/subctl-${VERSION}-linux-amd64.tar.xz

alias subctl=/tmp/subctl-${VERSION}/subctl-${VERSION}-linux-amd64
```

Reviewer: It can be simpler than that, the image can be used to run `subctl` directly:

    podman run --rm quay.io/submariner/subctl:devel subctl version

The main subtlety is that we need to add a volume mount to provide access to the kubeconfig.

See the https://submariner.io/getting-started/#prerequisites[Submariner upstream prerequisites documentation] for more detailed information about the prerequisites.

[#submariner-subctl]
== subctl command utility

Submariner contains the `subctl` utility that provides additional commands that simplify running tasks on your Submariner environment. After adding the utility to your path, the following table briefly describes the available commands:

|===
| https://submariner.io/operations/deployment/subctl/#export[`export`] | When `service` is added, creates a `ServiceExport` resource for the specified service, which enables other clusters in the Submariner deployment to discover the corresponding service. 

> | https://submariner.io/operations/deployment/subctl/#unexport[`unexport`] | When `service` is added, removes the `ServiceExport` resource for the specified service, which prevents other clusters in the Submariner deployment from discovering the corresponding service. 

Reviewer 2: I think it would make sense to treat these as `export service` and `unexport service`, as we only expect those args to be used together.

> | https://submariner.io/operations/deployment/subctl/#show[`show`] | Displays the list of different resources in a Submariner deployment.

Reviewer 2: ```suggestion
| https://submariner.io/operations/deployment/subctl/#show[`show`] | Provides information about Submariner resources.
```

> | https://submariner.io/operations/deployment/subctl/#verify[`verify`] | Tests and confirms that the Submariner deployment is working across the clusters.

Reviewer 2: ```suggestion
| https://submariner.io/operations/deployment/subctl/#verify[`verify`] | Verifies connectivity, service discovery, and other Submariner features across a pair of Submariner clusters.
```

> | https://submariner.io/operations/deployment/subctl/#benchmark[`benchmark`] | Runs benchmarking tests on throughput or latency in the clusters of a Submariner deployment. 

Reviewer 2: ```suggestion
| https://submariner.io/operations/deployment/subctl/#benchmark[`benchmark`] | Benchmarks throughput and latency across a pair of Submariner clusters or within a single cluster. 
```

| https://submariner.io/operations/deployment/subctl/#diagnose[`diagnose`] | Runs checks to identify issues that prevent the Submariner deployment from working correctly. 
| https://submariner.io/operations/deployment/subctl/#gather[`gather`] | Collects information from the clusters to help troubleshoot a Submariner deployment.

> | https://submariner.io/operations/deployment/subctl/#cloud[`cloud`] | Prepares the {ocp} installer-provisioned infrastructure (IPI) on the specified cloud providers. 

Reviewer 2: ```suggestion
```

I think this should not be covered, as clusters that support cloud-prepare should be prepared through the console, not manually.

See the https://submariner.io/getting-started/#prerequisites[Submariner upstream prerequisites documentation] for more detailed information about the prerequisites.

[#submariner-subctl]
== subctl command utility

Submariner contains the `subctl` utility that provides additional commands that simplify running tasks on your Submariner environment. 

[#Submariner-subctl-install]
=== Installing the subctl command utility

> The `subctl` utility is provided as a container image. Complete the following steps to install the `subctl` utility: 

Reviewer 2: ```suggestion
The `subctl` utility is shipped in a container image. Complete the following steps to install the `subctl` utility locally: 
```

Author: committed

> . Download and extract the container image by entering the following command: 

Reviewer 2: ```suggestion
. Download the `subctl` container and extract a compressed version of the `subctl` binary to `/tmp` by entering the following command: 
```

Author: Committed

+
----
oc image extract registry-proxy.engineering.redhat.com/rh-osbs/rhacm2-tech-preview-subctl-rhel8:${VERSION} --path=/dist/subctl-${VERSION}-linux-amd64.tar.xz:/tmp/ --confirm
----

> . Expand the file by entering the following command: 

Reviewer 2: ```suggestion
. Decompress `subctl` by entering the following command: 
```

Author: committed

+
----
tar -C /tmp/ -xf /tmp/subctl-${VERSION}-linux-amd64.tar.xz
----

> . Start the utility by entering the following command:
> +
> ----
> alias subctl=/tmp/subctl-${VERSION}/subctl-${VERSION}-linux-amd64
> ----

Reviewer 2: ```suggestion
. Install `subctl` by entering the following command:
+
----
install -m744 /tmp/subctl-${VERSION}/subctl-${VERSION}-linux-amd64 /$HOME/.local/bin/subctl
----
```

Author: committed
